src/node_messaging.cc

```diff
@@ -537,36 +544,87 @@ std::unique_ptr<MessagePortData> MessagePort::Detach() {
 }
 
 
-void MessagePort::Send(Message&& message) {
-  Mutex::ScopedLock lock(*data_->sibling_mutex_);
-  if (data_->sibling_ == nullptr)
-    return;
-  data_->sibling_->AddToIncomingQueue(std::move(message));
-}
+Maybe<bool> MessagePort::PostMessage(Environment* env,
+                                     Local<Value> message_v,
+                                     Local<Value> transfer_v) {
+  auto isolate = env->isolate();
+  auto obj = object(isolate);
+  auto context = obj->CreationContext();
+
+  std::vector<MessagePort*> transferred_ports;
+
+  // Check if the MessagePort itself is being transferred.
+  if (transfer_v->IsArray()) {
+    Local<Array> transfer_list = transfer_v.As<Array>();
+    uint32_t length = transfer_list->Length();
+    for (uint32_t i = 0; i < length; ++i) {
+      Local<Value> entry;
+      if (!transfer_list->Get(context, i).ToLocal(&entry))
```

Comment:
Can you add a comment for why this is needed?

Reply:
Hm, what do you mean? This is how to access array elements through the V8 API.